app/services/doc_auth/error_generator.rb

```diff
@@ -362,12 +385,28 @@ def scan_for_unknown_alerts(response_info)
       unknown_fail_count
     end
 
-    def self.general_error(_liveness_enabled)
-      Errors::GENERAL_ERROR
+    # This method replicates TrueIdResponse::attention_with_barcode? and
+    # should be removed/updated when that is.
+    def attention_with_barcode_result(doc_auth_result, processed_alerts)
+      attention_result = doc_auth_result == LexisNexis::ResultCodes::ATTENTION.name
+      barcode_alerts = processed_alerts[:failed]&.count.to_i == 1 &&
+                       processed_alerts.dig(:failed, 0, :name) == '2D Barcode Read' &&
+                       processed_alerts.dig(:failed, 0, :result) == 'Attention'
+
+      attention_result && barcode_alerts ? true : false
     end
 
-    def self.wrapped_general_error(liveness_enabled)
-      { general: [ErrorGenerator.general_error(liveness_enabled)], hints: true }
+    def doc_auth_passed_or_attn_with_barcode(response_info)
+      doc_auth_result = response_info[:doc_auth_result]
+      processed_alerts = response_info[:processed_alerts]
+
+      doc_auth_result_passed = doc_auth_result == LexisNexis::ResultCodes::PASSED.name
+      doc_auth_result_passed || attention_with_barcode_result(doc_auth_result, processed_alerts)
+    end
+
+    def doc_auth_error_count(response_info)
+      doc_auth_passed_or_attn_with_barcode(response_info) ?
```

Comment:
In reality, there are cases that `doc_auth_result` is `Passed` but the response has alerts and `product_status` is `Fail`. So the error generator will be invoked and the `alert` will not be accounted. But it indeed runs through the logic and ends up with a generic error.  Ideally it should be `alert` speicif error and it will require `selfie` re-upload even if `FaceMatchResult` is a `Pass`.

It may be out of scope here, but in future iterations especially when we switch to other criterias like using `transaction_status` for success.

Reply:
An example test
```ruby
it 'DocAuthResult is success with known alert and selfie success' do
  error_info = build_error_info(
    doc_result: 'Passed',
    failed: [{ name: '2D Barcode Read', result: 'Failed' }],
    )
  error_info[:liveness_enabled] = true
  # Selfie not match ID
  error_info[:portrait_match_results] = {
    FaceMatchResult: 'Pass',
    FaceErrorMessage: 'Successful. Liveness: Live',
  }

  output = described_class.new(config).generate_doc_auth_errors(error_info)

  expect(output.keys).to contain_exactly(:general, :front, :back, :hints, :selfie)
  expect(output[:hints]).to eq(false)
end
```

Reply:
I'm not sure I completely follow this case. To me looking at your test, your comment says `Selfie not match ID` but then `FaceMatchResult: 'Pass'`. As far as I know, that can never happen.

It does sound though like the level of effort in the change your describing might be a new ticket, as you suggested. Especially perhaps after (or when) we implement relying on `transaction_status`?

Maybe there is something I'm missing though.

Reply:
@night-jellyfish , ignore the comments, updated TC, mainly this kind of possible test data.

Reply:
I see. I guess in this case I would expect error generator to **not** ever be called due to [this line of code](https://github.com/18F/identity-idp/blob/main/app/services/doc_auth/lexis_nexis/responses/true_id_response.rb#L65).

Reply:
Ohh I see. Thanks for pointing that out! I think you are right that that is an bug that should be fixed.

I do think though that it should be a separate ticket at this point, given the scope of this change. And I don't think this change introduces that issue, I think it already existed (if I'm understanding correctly).

Reply:
Ticket created: https://cm-jira.usa.gov/browse/LG-13168